java/src/jmri/managers/DefaultIdTagManager.java

```diff
@@ -99,11 +101,6 @@ public char typeLetter() {
         return 'D';
     }
 
-    @Override
-    public String getSystemPrefix() {
-        return "I";
```

Comment:
For the IDTag Manager, the prefix needs to remain constant.  The same is true of RailCom and LocoNet transponding tags.

By the way, the prefix here actually comes from IDentification tag.  Similarly, RD is for Railcom iDentification and LD is for Loconet iDentification.

The key here is that the tags can’t change names just because we add another connection that can read them.

Reply:
No, it must match the System Connection to which it is tied. Since [2009](https://web.archive.org/web/20090307174118/http://www.jmri.org/help/en/html/doc/Technical/Names.shtml) (the farthest back I can search), JMRI has clearly documented that a System Name contains a prefix that allows JMRI to determine, from the name along, which connection handles the NamedBean (the earliest code I can find that actually does that is from [2010](https://github.com/JMRI/JMRI/commit/2ae821f6df4648a55678f0294b4c2d7036bcb627) (I don't know that more specific code for the same task existed elsewhere earlier)). Just because it is convenient to hijack a property with a conflicting meaning does mean it should be done.

If the user wants the default Internal connection prefix to be "z8" (lowercase Zulu eight) instead of "I" (uppercase India), all of JMRI  should respect that, and specific types should not be overriding that, unless it really is our intention that JMRI cannot match a namedBean to a connection.

So, I'm going to make `getSystemPrefix()` final in `AbstractNamedBean` and take steps to ensure that currently-internal only types that do not share the default internal prefix as of 4.16 get created with special unregistered memos. That should work without fault until such time as it is possible to provide a silent migration path to using prefixes that match the connected system.

Reply:
I don’t care if the prefix is fixed for all eternity.  

The change I have flagged here implies that there can be multiple DefaultIdTagManagers.  My concern about this stems from the tag objects being created automatically by the readers if they don’t already exist.  I.e. if there are two readers of a single type but one uses tag manager A and the other uses tag manager B, you end up with two tags that differ only in the system prefix when the tag is read by the two different readers.  We use the tags to keep track of ( at a minimum ) where pieces of equipment were last seen on the layout, so having two tags for the same physical object is problematic.

I think the Unregistered memo idea works for the three different IDTag variations ( plain IDTags, RailCom tags, and LocoNet IDTags ) as long as there is only ever one tag manager of a particular type registered.

Reply:
I have another suggestion. There could be a system letter for things that are "floating around". Things that doesn't belong to any connection but is shared amongs all the connections.

The system letter Y is not used and could symbolize that it's connected to three different systems.

Reply:
> I have another suggestion. There could be a system letter for things that are "floating around". Things that doesn't belong to any connection but is shared amongs all the connections.
> 
> The system letter E is not used and could symbolize that it's connected to three different systems.

That is what I really is for.... but remember we only have default prefixes.  

Also, one of the issues is that we have 3 different types of tags, so we need to use three different connection prefixes.  Having one additional one doesn’t help.